edbdeploy/projects/virtualbox.py

```diff
@@ -0,0 +1,505 @@
+                                    'barman': True,
+                                    'barman_server_private_ip': backup1['private_ip'],  # noqa
+                                    'barman_backup_method': 'postgres',
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+            inventory['all']['children'].update({
+                'standby': {
+                    'hosts': {}
+                }
+            })
+            for i in range(2, 4):
+                try:
+                    output = exec_shell(
+                        [
+                            self.bin("vagrant"),
+                            "ip",
+                            env.cloud_spec['postgres_server_%s' % i]['name']
+                        ],
+                        environ=self.environ,
+                        cwd=self.vagrant_project_path
+                    )
+                    result = output.decode("utf-8").split('\n')
+                    env.cloud_spec['postgres_server_%s' % i]['public_ip'] = result[0]  # noqa
+                    env.cloud_spec['postgres_server_%s' % i]['private_ip'] = result[0]  # noqa
+                except Exception as e:
+                    logging.error("Failed to execute the command")
+                    logging.error(e)
+                    raise CliError(
+                        ("Failed to obtain VirutalBox Instance IP Address for: %s,"
+                         "please check the logs for details.")
+                        % env.cloud_spec['postgres_server_%s' % i]['name']
+                    )
+        if env.reference_architecture == 'EDB-RA-3':
+            inventory['all']['children'].update({
+                'pgpool2': {
+                    'hosts': {}
+                }
+            })
+            for i in range(1, 4):
+                try:
+                    output = exec_shell(
+                        [
+                            self.bin("vagrant"),
+                            "ip",
+                            env.cloud_spec['pooler_server_%s' % i]['name']
+                        ],
+                        environ=self.environ,
+                        cwd=self.vagrant_project_path
+                    )
+                    result = output.decode("utf-8").split('\n')
+                    env.cloud_spec['pooler_server_%s' % i]['public_ip'] = result[0]  # noqa
+                    env.cloud_spec['pooler_server_%s' % i]['private_ip'] = result[0]  # noqa
+                except Exception as e:
+                    logging.error("Failed to execute the command")
+                    logging.error(e)
+                    raise CliError(
+                        ("Failed to obtain VirutalBox Instance IP Address for: %s,"
+                         "please check the logs for details.")
+                        % env.cloud_spec['pooler_server_%s' % i]['name']
+                    )
```

Comment:
I'd suggest you to rewrite this function by generating the machine list based on the reference architecture, then, iterate over each machine name and execute the `vagrant ip <machine-name>` command.